components/usage/pkg/scheduler/job.go

```diff
@@ -2,55 +2,72 @@
 // Licensed under the GNU Affero General Public License (AGPL).
 // See License-AGPL.txt in the project root for license information.
 
-package controller
+package scheduler
 
 import (
 	"context"
 	"fmt"
 	"github.com/gitpod-io/gitpod/common-go/log"
 	v1 "github.com/gitpod-io/gitpod/usage-api/v1"
+	"github.com/robfig/cron"
 	"google.golang.org/protobuf/types/known/timestamppb"
 	"time"
 )
 
-type Reconciler interface {
-	Reconcile() error
+type Job interface {
+	Run() error
 }
 
-type ReconcilerFunc func() error
+func NewLedgerTriggerJobSpec(schedule time.Duration, job Job) (JobSpec, error) {
+	parsed, err := cron.Parse(fmt.Sprintf("@every %s", schedule.String()))
+	if err != nil {
+		return JobSpec{}, fmt.Errorf("failed to parse ledger job schedule: %w", err)
+	}
 
-func (f ReconcilerFunc) Reconcile() error {
-	return f()
+	return JobSpec{
+		Job:      job,
+		ID:       "ledger",
+		Schedule: parsed,
+	}, nil
 }
 
-func NewLedgerReconciler(usageClient v1.UsageServiceClient, billingClient v1.BillingServiceClient) *LedgerReconciler {
-	return &LedgerReconciler{
+func NewLedgerTrigger(usageClient v1.UsageServiceClient, billingClient v1.BillingServiceClient) *LedgerJob {
+	return &LedgerJob{
 		usageClient:   usageClient,
 		billingClient: billingClient,
+
+		running: make(chan struct{}, 1),
```

Comment:
Why does the channel have a buffer size of `1`?

Reply:
Because we only want to run 1 at a time. This is in effect a mutex, either we can write into the channel (it's not full) and hence acquire the mutes, or we fail (and the `select` executes the `default` branch) in which case we didn't acquire the mutex.

Reply:
Ah, ok. I was getting confused. `1` is the default buffer size for a channel though, so it could be omitted here.

Reply:
Correct, I've included it just to be explicit, I'll also add a comment to this for future readers :)